profiling/nvtx-connector/kp_nvtx_connector.cpp

```diff
@@ -0,0 +1,190 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include <cstdio>
+#include <cstdint>
+#include <vector>
+#include <string>
+
+#include "nvToolsExt.h"
+
+#include "kp_core.hpp"
+
+static int tool_globfences;  // use an integer
```

Comment:
what is this used for? Was maybe the intend to use that in the Kokkos_request_tool_settings which currently is a no-op? And if so should the default be true?   If that is the case you can init it here to 1, and then you don't need a else clause in the init.

Reply:
I guess the goal is to allow the user to say "This profiling library requires fencing", using the environment variable `KOKKOS_TOOLS_GLOBALFENCES`. However, it seems other connectors don't have `KOKKOS_TOOLS_GLOBALFENCES`.

Reply:
Looks like this is now fixed good.

Reply:
Why is it an `int` and not a `bool`?

Reply:
> Why is it an `int` and not a `bool`?

My original reasoning was to provide extensibility to allow more modes of fencing. Talking with @crtrott, this is being changed to bool.